### ISSUE TITLE
Jason encoders

### ISSUE THREAD

**Issue by jc00ke** [wfgilman/plaid-elixir#14]
Right now we're having to derive the `Jason.Encoder` protocol on everything we need to serialize... not a huge deal, but since it's the default in Phoenix, it might be worth doing this by default.

What do you think?

This is what we have now:

```ex

require Protocol

Protocol.derive(Jason.Encoder, Plaid.Transactions.Transaction.Location)
Protocol.derive(Jason.Encoder, Plaid.Transactions.Transaction.PaymentMeta)
Protocol.derive(Jason.Encoder, Plaid.Institutions.Institution)
Protocol.derive(Jason.Encoder, Plaid.Institutions.Institution.Credentials)
Protocol.derive(Jason.Encoder, Plaid.Institutions.Institution.Colors)
```

**Comment by wfgilman** [comment]
Can you elaborate a bit about how the encoder protocol is needed for this library? Happy to accommodate.

**Comment by jc00ke** [comment]
We send these Plaid structs as JSON responses, and [by default Jason does not support encoding arbitrary structs](https://github.com/michalmuskala/jason/tree/cb1f26a8b3ca7efd4c79437b803cd8021c57d204#differences-to-poison). Since `Phoenix` and `Ecto` are moving/have moved away from `Poison` to `Jason`, I think it's worth at least calling out that one would have to manually derive `Jason.Encoder` if not providing some facility to do it _for_ the user.

**Comment by wfgilman** [comment]
Got it. I pushed the changes to a branch: https://github.com/wfgilman/plaid-elixir/pull/15

Wanna give it a look and see if it works for your use?

**Comment by jc00ke** [comment]
Yeah, that seems like it would cover my manual `derives`. Thanks!

Closed by #15
